### PR TITLE
Adding cytokine submodel to cc3d covid19 model

### PR DESCRIPTION
**Pull request by JulianoGianlupi**
The cytokine submodel is usable. I want to discuss it before pulling it into the main code.

Submodel verbal description: 

Immune cells may be inactivated (naive) or activated (trained to hunt invaders). All immune cells (activated or not) uptake cytokines (ck). After uptaking a certain amount (EC50, defined in the code) immune cells become activated. Activated immune cells chemotax on the ck field. They also secrete ck (at a rate 10x the rate of uptake). 

Infected (self.INFECTED) cells secrete cytokines (ck) into the environment (for now I am using the same same rate as activated immune, I haven't found an experimental number for them yet).

Cytokines **do not** decay, they are only uptaken. With the simulation as is ck diffuses way faster than virus, whole lattice field becomes same value fast. According to Josh virus D is lower than in reality, it should be closer to ck D. The way to solve the saturation of the field might be to increase lattice and diminish how many um is the side of a pixel.  

TBD: 

- replace instant activation for a probability of activation that depends on amount of ck uptaken (most likely a Hill)
- have some decay on the amount of ck uptaken (otherwise immune cells will have a memory that goes back to the beginning of Time)
- find a good ck secretion rate for infected cells
- replace constant rate of secretion/absorption for something that looks sigmoidal (don't know what would be the controlling variable; density of virus inside cytoplasm for the infected seems logical, not sure what could be a good one for activated immunes)
- fine tuning of ck diffusion constants. 

Do all of you feel that it is in an ok state to be part of the main code?

### CODE REVIEW

**Comment by JulianoGianlupi** [comment]
Also tbd add a way for immune to deactivate

**Review by tjsego — APPROVED** [review]
My only issue is no cytokine decay. However, this is something we can easily explore, so I think it's a non-issue for now. This is a great set of additions to the model. We'll refine and clean as we go.

**Comment by tjsego** [comment]
Also, let's merge this branch with master once this is approved. Docs are in better shape but need merged for image links to work properly.

**Pull request merged**
